apps/studio/components/interfaces/Realtime/RealtimeSettings.tsx

```diff
@@ -0,0 +1,345 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import { useEffect } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { toast } from 'sonner'
+import * as z from 'zod'
+
+import { useParams } from 'common'
+import { ScaffoldSection } from 'components/layouts/Scaffold'
+import { convertFromBytes } from 'components/to-be-cleaned/Storage/StorageSettings/StorageSettings.utils'
+import AlertError from 'components/ui/AlertError'
+import { FormSection, FormSectionContent, FormSectionLabel } from 'components/ui/Forms/FormSection'
+import { useRealtimeConfigurationUpdateMutation } from 'data/realtime/realtime-config-mutation'
+import {
+  REALTIME_DEFAULT_CONFIG,
+  useRealtimeConfigurationQuery,
+} from 'data/realtime/realtime-config-query'
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import {
+  Button,
+  Card,
+  CardContent,
+  CardFooter,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Input_Shadcn_,
+  Switch,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+
+const formId = 'realtime-configuration-form'
+
+const FormSchema = z.object({
+  private_only: z.boolean(),
+  connection_pool: z.number().min(1).max(100),
+  max_concurrent_users: z.number().min(1).max(50000),
+  max_events_per_second: z.number().min(1).max(50000),
+  max_bytes_per_second: z.number().min(1).max(10000000),
+  max_channels_per_client: z.number().min(1).max(10000),
+  max_joins_per_second: z.number().min(1).max(5000),
+})
+
+export const RealtimeSettings = () => {
+  const { ref: projectRef } = useParams()
+  const canUpdateConfig = useCheckPermissions(PermissionAction.REALTIME_ADMIN_READ, '*')
+
+  const { data, error, isLoading, isSuccess, isError } = useRealtimeConfigurationQuery({
+    projectRef,
+  })
+  const { mutate: updateRealtimeConfig, isLoading: isUpdatingConfig } =
+    useRealtimeConfigurationUpdateMutation({
+      onSuccess: () => {
+        form.reset(form.getValues())
```

Comment:
Why is this `reset` needed?

Reply:
yup this is so that the form's default values get updated (and the cancel button goes away + save button disables)